source/Halibut/Transport/TcpConnectionFactory.cs

```diff
@@ -47,12 +47,12 @@ public IConnection EstablishNewConnection(ExchangeProtocolBuilder exchangeProtoc
             return new SecureConnection(client, ssl, exchangeProtocolBuilder, log);
         }
         
-        public async Task<IConnection> EstablishNewConnectionAsync(ExchangeProtocolBuilder exchangeProtocolBuilder, ServiceEndPoint serviceEndpoint, ILog log, CancellationToken cancellationToken)
+        public async Task<IConnection> EstablishNewConnectionAsync(ExchangeProtocolBuilder exchangeProtocolBuilder, ServiceEndPoint serviceEndpoint, HalibutTimeoutsAndLimits halibutTimeoutsAndLimits, ILog log, CancellationToken cancellationToken)
```

Comment:
Given the limits are set on the runtime, should they be given to the constructor of the factory and then used in the methods so wee don't have to pass them around to all methods

Reply:
May mean some methods below would benefit from not being static